migrate/20241014_add_connected_subnet.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  change do
+    create_table :connected_subnet do
+      column :id, :uuid, primary_key: true, null: false
+      foreign_key :subnet_id_1, :private_subnet, type: :uuid, null: false
+      foreign_key :subnet_id_2, :private_subnet, type: :uuid, null: false
+
+      # Ensure no duplicate pairs regardless of order
+      constraint(:unique_subnet_pair) { (subnet_id_1 < subnet_id_2) }
+      unique [:subnet_id_1, :subnet_id_2]
+      index [:subnet_id_1]
```

Comment:
you probably don't need this one, because the compound index can handle predicates on the left-most-side. It's not quite the same physical impact, since the locality of scanning on the compound index would be worse, but would make updates a bit cheaper.